module/executiondatasync/optimistic_sync/errors.go

```diff
@@ -0,0 +1,75 @@
+package optimistic_sync
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// RequiredExecutorsCountExceededError indicates that the requested number of required executors
+// exceeds the total available execution nodes.
+type RequiredExecutorsCountExceededError struct {
+	requiredExecutorsCount  int
+	availableExecutorsCount int
+}
+
+func NewRequiredExecutorsCountExceededError(requiredExecutorsCount int, availableExecutorsCount int) *RequiredExecutorsCountExceededError {
+	return &RequiredExecutorsCountExceededError{
+		requiredExecutorsCount:  requiredExecutorsCount,
+		availableExecutorsCount: availableExecutorsCount,
+	}
+}
+
+func (e *RequiredExecutorsCountExceededError) Error() string {
```

Comment:
it's a good idea to also include `Unwrap` methods to allow `errors.Is` and `errors.As` to work when the returned error is wrapped.